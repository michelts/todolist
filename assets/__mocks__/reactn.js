export const useGlobal = jest.fn();
