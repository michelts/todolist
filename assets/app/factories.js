import Factory from 'rosie';

const UserFactory = new Factory()
  .sequence('id', (index) => index + 1)
  .sequence('username', (index) => `username_${index}`)
  .sequence('name', (index) => `Name-${index}`);

export {
  UserFactory,
};
