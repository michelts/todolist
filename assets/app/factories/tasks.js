import { Factory } from 'rosie';

const TaskFactory = new Factory()
  .sequence('id', (index) => index + 1)
  .sequence('description', (index) => `Task ${index}`)
  .attr('priority', 1)
  .attr('due_date', '2020-05-22')
  .attr('is_completed', false);

export {
  TaskFactory,
}
