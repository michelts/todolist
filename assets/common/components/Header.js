import React from 'react'
import Navbar from 'react-bootstrap/Navbar';

const Header = ({ children }) => (
  <Navbar bg="dark" variant="dark">
    <Navbar.Brand href="#home">{children}</Navbar.Brand>
  </Navbar>
);


export default Header;
