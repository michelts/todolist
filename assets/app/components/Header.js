import React from 'react';
import { useGlobal } from 'reactn';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';

const Header = ({ children }) => {
  const [user] = useGlobal('user');
  return (
    <Navbar
      bg="dark"
      variant="dark"
      className="justify-content-between"
    >
      <Navbar.Brand href="#/">
        {children}
      </Navbar.Brand>
      {user && (
        <Navbar.Collapse
          className="justify-content-end"
        >
          <Nav.Link href="#/logout">Logout</Nav.Link>
        </Navbar.Collapse>
      )}
    </Navbar>
  );
};

export default Header;
