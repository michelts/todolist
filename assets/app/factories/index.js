import { UserFactory } from './user';
import { TaskFactory, BlankTaskFactory } from './tasks';

export {
  UserFactory,
  TaskFactory,
  BlankTaskFactory,
};
