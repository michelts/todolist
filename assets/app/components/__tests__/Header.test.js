import React from 'react';
import { useGlobal } from 'reactn';
import { shallow } from 'enzyme';
import { UserFactory } from '../../factories';
import Header from '../Header';

const getWrapper = () => (
  shallow(<Header>Brand name</Header>)
);

describe('Header component', () => {
  it('should render the brand with given children, if user is undefined', () => {
    useGlobal.mockReturnValue([undefined]);
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the brand with given children, if user is not authenticated', () => {
    useGlobal.mockReturnValue([null]);
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the brand and a logout link, if user is authenticated', () => {
    useGlobal.mockReturnValue([UserFactory.build()]);
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });
});
