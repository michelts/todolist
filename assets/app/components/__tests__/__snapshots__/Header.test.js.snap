// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render the brand and a logout link, if user is authenticated 1`] = `
<Navbar
  bg="dark"
  className="justify-content-between"
  collapseOnSelect={false}
  expand={true}
  variant="dark"
>
  <NavbarBrand
    href="#/"
  >
    Brand name
  </NavbarBrand>
  <NavbarCollapse
    className="justify-content-end"
  >
    <NavLink
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "SafeAnchor",
          "render": [Function],
        }
      }
      disabled={false}
      href="#/logout"
    >
      Logout
    </NavLink>
  </NavbarCollapse>
</Navbar>
`;

exports[`Header component should render the brand with given children, if user is not authenticated 1`] = `
<Navbar
  bg="dark"
  className="justify-content-between"
  collapseOnSelect={false}
  expand={true}
  variant="dark"
>
  <NavbarBrand
    href="#/"
  >
    Brand name
  </NavbarBrand>
</Navbar>
`;

exports[`Header component should render the brand with given children, if user is undefined 1`] = `
<Navbar
  bg="dark"
  className="justify-content-between"
  collapseOnSelect={false}
  expand={true}
  variant="dark"
>
  <NavbarBrand
    href="#/"
  >
    Brand name
  </NavbarBrand>
</Navbar>
`;
